### PR TITLE
Add integration tests for neighbors

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Adds integration tests for neighbor functionality

## ⚠️ Known issues

- Default distance function is currently not set to "euclidean"

## 🐾 Next steps

- Change default distance function to "euclidean"
- Implement more integration tests

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201481007343159/1201671951990191/f) (_internal_)
- [Wrong distance function](https://app.asana.com/0/1201481007343159/1201674703280810/f) (_internal_)

## 🛡 What tests cover this?

- This are the tests

## ❓ How to test this?

- run `cargo test`

## 📹 Demo
image

### CODE REVIEW

File: packages/engine/tests/units/neighbors/fields/bool/src/behaviors/test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Accesses field `value`
+ */
+const behavior = (state, context) => {
+  const neighbor = context.neighbors()[0];
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/engine/tests/units/neighbors/fields/bool/src/behaviors/test.js:5]: *If* we choose to extend the tests like this, lets make sure we capture those assumptions somehow (i.e. the ordering etc.) to explain the reasoning of the test

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/engine/tests/units/neighbors/fields/bool/src/behaviors/test.js:5]: Good point!
I'd add one test with multiple agents so we can check this without polluting every test with multiple agents.

[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/engine/tests/units/neighbors/fields/bool/src/behaviors/test.js:5]: Oh valid, yeah you're right, no need to add it to every test. We could possibly integrate it into the search radius test but I think perhaps having it as its own will also help with that communication

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/engine/tests/units/neighbors/fields/bool/src/behaviors/test.js:5]: Added a test for it.

**Pull request merged**